src/wally-connector.ts

```diff
@@ -234,96 +243,168 @@ class WallyConnector {
+            : {}),
         },
+        ...(params &&
+          params.length > 0 && {
+            body: this.formatWallyParams(method, params),
+          }),
       });
       if (resp && resp?.ok && resp?.status < 300) {
         const data = await resp.json();
-        this.selectedAddress = data.address;
-        return [this.selectedAddress || ''];
+        return this.formatWallyResponse(method, data);
       } else {
         console.error(
-          'The Wally server returned a non-successful response when fetching wallet details'
+          `The Wally server returned a non-successful response when handling method: ${method}`
         );
-        await this.loginWithEmail();
       }
     } catch (err) {
-      console.error(`Unable to fetch Wally wallet: ${err}`);
-    }
-    return [];
-  }
-
-  async signMessage(params: string[]): Promise<SignedMessage | string> {
-    const resp = await fetch(`${this.host}/oauth/wallet/sign-message`, {
-      method: 'POST',
-      headers: {
-        Authorization: `Bearer ${this.getAuthToken()}`,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        message: params[1],
-      }),
-    });
-
-    if (!resp.ok || resp.status >= 300) {
-      throw new Error(
-        'Wally server returned a non-successful response when signing a message'
+      console.error(
+        `Wally server returned error: ${err} when handling method: ${method}`
       );
     }
-    const json = await resp.json();
-    return json.signature;
+
+    return null;
   }
 
   /**
    * Handle other non-wally-specific methods - forwards to ethers/alchemy
    * on the backend
-   * @param method The RPC Method
-   * @param params Arbitrary array of params
-   * @returns whatever you want it to
+   * @param method The RPC method name
+   * @param params The json rpc spec params
+   * @returns RPCResponse - adheres to the json rpc spec
    */
-  private async _request(method: string, params: string[]): Promise<any> {
-    const resp = await fetch(`${this.host}/oauth/wallet/send`, {
-      method: 'POST',
-      headers: {
-        Authorization: `Bearer ${this.getAuthToken()}`,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        method,
-        params,
-      }),
-    });
+  private async requestRPC<T extends RPCMethodName>(
+    method: T,
+    params: RPCMethodParams<T>
+  ): Promise<RPCResponse<T> | null> {
+    console.log({ method, params });
+    try {
+      const resp = await fetch(`${this.host}/oauth/wallet/send`, {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.getAuthToken()}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          method,
+          params,
+        }),
+      });
+
+      if (!resp.ok || resp.status >= 300) {
+        throw new Error(
+          'Wally server returned a non-successful response when signing a message'
```

Comment:
does this error message also need to be updated?

Reply:
good catch!